lib/utils.js

```diff
@@ -506,6 +503,40 @@ exports.canonicalize = function canonicalize(value, stack, typeHint) {
   return canonicalizedObj;
 };
 
+/**
+ * Determines if pathname has a matching file extension.
+ *
+ * @private
+ * @param {string} pathname - Pathname to check for match.
+ * @param {string[]} exts - List of file extensions (sans period).
+ * @return {boolean} whether file extension matches.
+ * @example
+ * hasMatchingExtname('foo.html', ['js', 'css']); // => false
+ */
+function hasMatchingExtname(pathname, exts) {
```

Comment:
is the regexp really necessary here?  I know you just moved this code, but:

```js
function hasMatchingExtname(pathname, exts) {
  return exts.indexOf(path.extname(pathname).slice(1)) > -1;
}
```

Reply:
Made a [JSPerf](https://jsperf.com/hasmatchingextname) for this. Had started down this route, but a different performance check showed seesawing results across different browser-version combinations.

Reply:
Should the matching extname check have been done case-insensitively?

Reply:
I'd say no.  never seen a `.JS` file in the wild.

Reply:
either way, it's not hugely important